lib/internal/Magento/Framework/App/State.php

```diff
@@ -135,6 +145,10 @@ public function setAreaCode($code)
         }
         $this->_configScope->setCurrentScope($code);
         $this->_areaCode = $code;
+
```

Comment:
That's what have been done in [lib/internal/Magento/Framework/App/StaticResource.php](https://github.com/magento/magento2/pull/37727/files#diff-0fda5f9b0cea0b50ba4e9f1922be572b6c46130c7f183a99e3f008fad65355b1) ... But having to add this code after each time we call setAreaCode() to avoid errors seems wrong to me ?